lib/network/rest/server.js

```diff
@@ -0,0 +1,332 @@
+}
+
+function setContentRange(response, byteRange, objectSize) {
+    const [start, end] = byteRange;
+    assert(start !== undefined && end !== undefined);
+    response.setHeader('Content-Range',
+                       `bytes ${start}-${end}/${objectSize}`);
+}
+
+function sendError(res, code, message) {
+    let httpCode;
+    if (typeof(code) === 'number' && code !== 0) {
+        httpCode = code;
+    } else {
+        httpCode = 500;
+    }
+    res.writeHead(httpCode);
+    if (message) {
+        res.end(JSON.stringify({ message }));
+    } else {
+        res.end();
+    }
+}
+
+function createLoggingObject(config) {
+    let options = undefined;
+    if (config !== undefined) {
+        options = {
+            level: config.logLevel,
+            dump: config.dumpLevel,
+        };
+    }
+    return new Logger('RESTServer', options);
+}
+
+/**
+ * @class
+ * @classdesc REST Server interface
+ *
+ * You have to call setup() to initialize the storage backend, then
+ * start() to start listening to the configured port.
+ */
+class RESTServer extends httpServer {
+
+    /**
+     * @constructor
+     * @param {Object} params - constructor params
+     * @param {Number} params.port - TCP port where the server listens to
+     * @param {Object} [params.log] - logger configuration
+     */
+    constructor(params) {
+        assert(params.port);
+
+        const logging = createLoggingObject(params.log);
+        super(params.port, logging);
+        this.logging = logging;
+        this.dataStore = params.dataStore;
+
+        this.onRequest(this._onRequest);
+    }
+
+    /**
+     * Setup the storage backend
+     *
+     * @param {function} callback - called when finished
+     * @return {undefined}
+     */
+    setup(callback) {
+        this.dataStore.setup(callback);
+    }
+
+    /**
+     * Create a new request logger object
+     *
+     * @param {String} reqUids - serialized request UIDs (as received in
+     * the X-Scal-Request-Uids header)
+     * @return {Logger} new request logger
+     */
+    createLogger(reqUids) {
+        return reqUids ?
+            this.logging.newRequestLoggerFromSerializedUids(reqUids) :
+            this.logging.newRequestLogger();
+    }
+
+    /**
+     * Main incoming request handler, dispatches to method-specific
+     * handlers
+     *
+     * @param {http.IncomingMessage} req - HTTP request object
+     * @param {http.ServerResponse} res - HTTP response object
+     * @return {undefined}
+     */
+    _onRequest(req, res) {
+        const reqUids = req.headers['x-scal-request-uids'];
+        const log = this.createLogger(reqUids);
+        log.debug('request received', { method: req.method,
+                                        url: req.url });
+        const reqMethods = {
+            PUT: this._onPut.bind(this),
+            GET: this._onGet.bind(this),
+            DELETE: this._onDelete.bind(this),
+        };
+        if (req.method in reqMethods) {
```

Comment:
Using `in` may not be the only option. For this kind of fixed corpus, you can also use `switch ... case` so that you don't have to create the object `reqMethods` every single request.
The `bind` is not needed here neither btw.